nixos/lib/make-disk-image.nix

```diff
@@ -302,9 +302,23 @@ in pkgs.vmTools.runInLinuxVM (
         cp ${configFile} /mnt/etc/nixos/configuration.nix
       ''}
 
+
+      ${optionalString (partitionTableType == "efi" || partitionTableType == "hybrid") ''
+        # This is needed for systemd-boot to find ESP from inside runInLinuxVM
+        mkdir /dev/block
+        ln -s /dev/vda1 /dev/block/254:1
+      ''}
+
       # Set up core system link, GRUB, etc.
       NIXOS_INSTALL_BOOTLOADER=1 nixos-enter --root $mountPoint -- /nix/var/nix/profiles/system/bin/switch-to-configuration boot
 
+      ${optionalString (partitionTableType == "efi" || partitionTableType == "hybrid") ''
+        # remove symlinks required only inside runInLinuxVM again
+        rm /dev/block/254:1
+        rmdir /dev/block
+      ''}
+
+
```

Comment:
And I believe it would be fine to keep it, no need to remove it.

Makes the script easier to maintain. And the build VM will look the same no matter what.

After all, this emulates what a `udev` like program would have done. It generally won't undo what it does after the bootloader has been installed.